torch/csrc/jit/test_jit.cpp

```diff
@@ -484,8 +484,8 @@ std::shared_ptr<Graph> trace(const ADTestSpec& test, const variable_list& vars_i
 }
 
 variable_list grad(const variable_list& outputs, const variable_list& inputs, const variable_list& grad_outputs) {
-  static const auto get_edge = [](const Variable& v) -> edge_type {
-    return std::make_pair(v.grad_fn() ? v.grad_fn() : v.grad_accumulator(), v.output_nr());
+  static const auto get_edge = [](const Variable& v) {
+    return FunctionPort(v.grad_fn() ? v.grad_fn() : v.grad_accumulator(), v.output_nr());
```

Comment:
Another place where the same pattern similar to `get_gradient` appears, but without checks, we should have a function.